src/CornerstoneViewport/CornerstoneViewport.js

```diff
@@ -45,6 +45,9 @@ function initializeTools(cornerstoneTools, tools, element) {
 }
 
 function layoutsEqual(a, b) {
+  if (!a || !b || !a.viewports || !b.viewports) {
+    return false;
+  }
```

Comment:
We may want this to return `true` if one `viewports` is undefined, and the other is not? But yes, if both are undefined, we should definitely return false.

Reply:
done

Reply:
The trick is, if only one `viewports` is undefined ;)

```
if(!a || !b || (!a.viewports && !b.viewports)) {
  return false;
}
```

Then in the second conditional,

```
if( !a.viewports || !b.viewports || a.viewports.length !== b.viewports.length)
```